packages/devtools/src/elementstore.ts

```diff
@@ -25,7 +36,16 @@ export default class ElementStore {
         return isElementAttachedToDOM ? elementHandle : undefined
     }
 
-    clear () {
-        this._elementMap.clear()
+    clear (frame?: Frame) {
+        if (!frame) {
+            this._elementMap.clear()
+            this._frameMap.clear()
+        } else {
+            const elementIndexes = this._frameMap.get(frame)
+            if (elementIndexes) {
+                elementIndexes.forEach((elementIndex) => this._elementMap.delete(elementIndex))
+                this._frameMap.delete(frame)
+            }
+        }
```

Comment:
This is just personal preference, feel free to ignore but
```suggestion
        if (!frame) {
            this._elementMap.clear()
            this._frameMap.clear()
            return
        }

        const elementIndexes = this._frameMap.get(frame)
        if (elementIndexes) {
            elementIndexes.forEach((elementIndex) => this._elementMap.delete(elementIndex))
            this._frameMap.delete(frame)
        }
```